paper/paper.md

```diff
@@ -0,0 +1,201 @@
+pyGeo contains an implementation of this airfoil parameterization that supports design variables for the Bernstein polynomial weightings, the class function parameters, and the airfoil chord length.
+pyGeo's CST implementation can only be used for 2D problems, such as airfoil optimization (\autoref{fig:cst_example}).
+
+![Airfoil defined by three CST coefficients on each surface undergoing a perturbation in one Bernstein polynomial. \label{fig:cst_example}](cst_example.pdf){ width=85% }
+
+## Constraints
+pyGeo also includes geometric constraints to prevent geometrically undesirable designs.
+The most commonly-used class of geometry constraints in pyGeo involves tracking one or more linear dimensions on the optimized object's surface.
+These constraints are created by specifying a single point, a line, or an array of points, along with a normal direction, then computing two line-surface intersection points.
+Some commonly used geometric constraints in shape optimization, such as thickness, area, and volume constraints (\autoref{fig:constraint}) can be computed using variations on this approach, which is computationally cheap and robust [@Brelje2020a].
+
+![Thickness and volume constraints demonstrated on an wing section [@Brelje2020a]. \label{fig:constraint}](constraints_3d.pdf){ width=75% }
+
+If a more complex geometry needs to be integrated into an optimized surface, pyGeo supports an alternative geometric constraint formulation based on arbitrary triangulated surfaces as illustrated in \autoref{fig:trisurf} [@Brelje2020a].
+
+![Triangulated surface constraint used to optimize an aeroshell around a complex geometry [@Brelje2020a].\label{fig:trisurf}](trisurfcon.pdf)
+
+
+# Parallelism
+pyGeo can optionally work under distributed memory parallelism using MPI, which is required when interfacing with large-scale CFD applications.
+For example, the computational mesh may be partitioned and distributed among many processors by the CFD solver, and each processor may be aware of only its portion of the mesh.
+pyGeo can handle such scenarios by independently manipulating the geometry on each processor and aggregating the constraints across all processors when communicating with the optimizer.
+
+
+# Derivative Computation
+In addition to geometry manipulation and constraints, pyGeo can compute derivatives of these operations with respect to design variables.
+For the geometric deformation, pyGeo can compute the Jacobian matrix
+\begin{equation*}
+\frac{\mathrm{d}X_s}{\mathrm{d}x} ,
+\end{equation*}
+where $X_s$ is the vector of surface mesh coordinates, and $x$ is the vector of geometric design variables.
+
+Similarly, pyGeo can compute the constraint Jacobian matrix
+\begin{equation*}
+\frac{\mathrm{d}g}{\mathrm{d}x} ,
+\end{equation*}
+where $g$ is the vector of geometric constraints.
+
+For the FFD parameterization, these derivatives are computed using implicit analytic methods [@Martins2021, Sec. 6.7] and the complex-step method [@Martins2003a].
```

Comment:
One other point that I missed earlier: I don't think "implicit analytic methods" is correct. This section of the book refers to the direct and adjoint methods, but the FFD parameterization does not solve any linear systems. The FFD parameterization is an explicit function that is differentiated mostly by hand (and some complex step).